apps/login/app/(login)/register/page.tsx

```diff
@@ -3,24 +3,46 @@ import {
   getPasswordComplexitySettings,
   server,
 } from "#/lib/zitadel";
-import RegisterForm from "#/ui/RegisterForm";
+import RegisterFormWithoutPassword from "#/ui/RegisterFormWithoutPassword";
+import SetPasswordForm from "#/ui/SetPasswordForm";
+
+export default async function Page({
+  searchParams,
+}: {
+  searchParams: Record<string | number | symbol, string | undefined>;
+}) {
+  const { firstname, lastname, email } = searchParams;
+
+  const setPassword = !!(firstname && lastname && email);
```

Comment:
Shouldn't the user either select to set a password or to register w/o pw?
What's the idea behind these criteria?

Reply:
thanks for clarifying :+1: